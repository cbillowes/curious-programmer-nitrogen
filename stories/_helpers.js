import React from 'react'
import Layout from '../src/components/layout'
import { SmallSpacer } from '../src/components/spacer'

const Page = ({ className, children }) => {
  return (
    <Layout
      className={className}
    >
      <SmallSpacer />
      {children}
    </Layout>
  )
}

export default Page