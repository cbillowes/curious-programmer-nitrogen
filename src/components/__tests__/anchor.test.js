import React from 'react'
import Anchor from '../anchor'
import { validateValues, getSnapshot } from './_helpers'

describe(`Anchor`, () => {
  it(`should link to http/https`, () => {
    const tree = getSnapshot(
      <Anchor
        to="https://curiousprogrammer.dev"
      >
        Curious Programmer
      </Anchor>
    )
    expect(tree).toMatchSnapshot()
  })

  it(`link to mailto`, () => {
    const tree = getSnapshot(
      <Anchor
        to="mailto:clarice@bouwer.dev"
      >
        clarice@bouwer.dev
      </Anchor>
    )
    expect(tree).toMatchSnapshot()
  })

  it(`link internally`, () => {
    const tree = getSnapshot(
      <Anchor
        to="/about"
      >
        About
      </Anchor>
    )
    expect(tree).toMatchSnapshot()
  })
})
