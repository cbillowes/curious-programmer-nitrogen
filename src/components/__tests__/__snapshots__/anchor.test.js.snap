// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Anchor link internally 1`] = `
Array [
  " ",
  <a
    href="/about"
    onClick={[Function]}
    onMouseEnter={[Function]}
  >
    About
  </a>,
  " ",
]
`;

exports[`Anchor link to mailto 1`] = `
Array [
  " ",
  <a
    href="mailto:clarice@bouwer.dev"
    rel="nofollow noopener noreferrer"
    target="_blank"
  >
    clarice@bouwer.dev
  </a>,
  " ",
]
`;

exports[`Anchor should configure class name externally 1`] = `
Array [
  " ",
  <a
    className="some-or-other-class"
    href="https://curiousprogrammer.dev"
    rel="nofollow noopener noreferrer"
    target="_blank"
  >
    Curious Programmer
  </a>,
  " ",
]
`;

exports[`Anchor should configure class name internally 1`] = `
Array [
  " ",
  <a
    className="some-or-other-class"
    href="/about"
    onClick={[Function]}
    onMouseEnter={[Function]}
  >
    About
  </a>,
  " ",
]
`;

exports[`Anchor should link to http/https 1`] = `
Array [
  " ",
  <a
    href="https://curiousprogrammer.dev"
    rel="nofollow noopener noreferrer"
    target="_blank"
  >
    Curious Programmer
  </a>,
  " ",
]
`;
